pkg/operator2/operator.go

```diff
@@ -330,12 +333,19 @@ func (c *authOperator) handleSync(operatorConfig *operatorv1.Authentication) err
 		operatorDeployment,
 		resourceVersions...,
 	)
+
+	// redeploy on operatorConfig.spec changes or when bootstrap user is deleted
+	forceRollOut := (operatorConfig.Generation != operatorConfig.Status.ObservedGeneration)
+	if !c.bootstrapUserExists() && c.bootstrapUserChangeRollOut {
+		forceRollOut = true
+		c.bootstrapUserChangeRollOut = !c.bootstrapUserChangeRollOut
```

Comment:
this will cause an unnecessary rollout every time the operator is started anew after the bootstrap user has been removed

Reply:
That is a good point which I failed to think when implementing this.

Now I thought about it a little bit and this is what I could come up with. Instead of blindly setting it to true, `bootstrapUserChangeRollOut` is set conditionally, thereby avoiding the forced roll outs.